plugin/role_set.go

```diff
@@ -415,6 +416,18 @@ func roleSetServiceAccountName(rsName string) (name string) {
 	return name
 }
 
+func roleSetServiceAccountDisplayName(name string) string {
+	hashLen := 8
+	fullDisplayName := fmt.Sprintf(serviceAccountDisplayNameTmpl, name)
+	displayName := fullDisplayName
+	if len(fullDisplayName) > serviceAccountDisplayNameMaxLen {
+		truncIndex := serviceAccountDisplayNameMaxLen - hashLen
+		h := fmt.Sprintf("%x", sha256.Sum256([]byte(fullDisplayName[truncIndex:])))
```

Comment:
Should the sha256sum be calculated off the full `fullDisplayName` to reduce the chances of a collision or is there a particular reason we're only considering bytes within `fullDisplayName[truncIndex:]`? The final length on the display name seems to be truncated by `h[:hashLen]` anyways.

Reply:
Actually, the first part of the display name is constructed via `fullDisplayName[:truncIndex]` so even if the hash ends up being the same as another one that had the same suffix string, the first part would be unique if the full display name is different from the one that it "collided" with so I think it's fine.